### ISSUE TITLE
event-item is broken in Inventory Click Event

### ISSUE THREAD

**Issue by MinecraftMan1013number2** [SkriptLang/Skript#5016]
### Skript/Server Version

```markdown
[16:37:46 INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[16:37:46 INFO]: [Skript] Skript's documentation can be found here: https://docs.skriptlang.org/
[16:37:46 INFO]: [Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials
[16:37:46 INFO]: [Skript] Server Version: git-Paper-81 (MC: 1.19)
[16:37:46 INFO]: [Skript] Skript Version: 2.6.3
[16:37:46 INFO]: [Skript] Installed Skript Addons: None
[16:37:46 INFO]: [Skript] Installed dependencies: None
```

### Bug Description

If I use `event-item` in an inventory click event, it always returns <none>.

### Expected Behavior

It broadcasts whatever the result is.

### Steps to Reproduce

To test this, I used both pieces of code below. I clicked on a diamond sword with no name (which is expected to return none), but then I clicked on a sword that I renamed in an anvil and it still broadcasted none.
```
on inventory click:
	broadcast "%name of event-item%"
```
```
on inventory click:
	broadcast "%event-item%"
```

### Errors or Screenshots

None

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

**Comment by AyhamAl-Ali** [comment]
I can't reproduce (both skript renamed and anvil renamed work, even with diamond sword)
![image](https://user-images.githubusercontent.com/20037329/183616980-a6953f0d-8f23-4c2a-85e2-7ec6da5e6401.png)

**Comment by MinecraftMan1013number2** [comment]
I also added a named item with color, and still the same
https://user-images.githubusercontent.com/70504198/183656041-cd4fbb33-c819-4187-a979-58de569c6037.mp4

**Comment by AyhamAl-Ali** [comment]
Seems like something is affect thing, do you have any other scripts running?

**Comment by MinecraftMan1013number2** [comment]
no, this is a blank server.

**Comment by AyhamAl-Ali** [comment]
Can you upload the whole  server as a .zip here

**Comment by Mythochu** [comment]
This also happened to me, it definitely isn't a server issue because I just went back to v2.6.2, and event-item started working again.

**Comment by MinecraftMan1013number2** [comment]
> Can you upload the whole server as a .zip here

https://file.io/hmpyaBzcPLoT

**Comment by wallace4BR** [comment]
maybe wait a tick before getting event-item? iirc, event-inventory returns null if doesn't

**Comment by MinecraftMan1013number2** [comment]
> maybe wait a tick before getting event-item? iirc, event-inventory returns something different if doesn't wait

Still the same result for me

**Comment by kimjunsung04** [comment]
I also had the same problem.
Any solution?

**Comment by kimjunsung04** [comment]
https://github.com/SkriptLang/Skript/commit/0372ad611477b08a1b285bf8d655e643ad19b086#diff-ab27ab8b1311587b5c719143d0bd197a4e311e0dd073ba820f04b608b9187ee4
I'm guessing, but it looks like something went wrong with that commit.

**Comment by TheLimeGlass** [comment]
I cannot replicate. Is everyone using ViaVersion? If so what version is your client?

On a 1.19.2 client and 1.19.2 server with or without ViaVersion I cannot replicate.

ViaVersion is designed for lower minecraft versions like 1.9 server to add 1.9+ support. It really serves no purpose being on a server that is the latest (1.19.2) there is even a warning in console from ViaVersion stating this.

**Comment by TheLimeGlass** [comment]
> [0372ad6#diff-ab27ab8b1311587b5c719143d0bd197a4e311e0dd073ba820f04b608b9187ee4](https://github.com/SkriptLang/Skript/commit/0372ad611477b08a1b285bf8d655e643ad19b086#diff-ab27ab8b1311587b5c719143d0bd197a4e311e0dd073ba820f04b608b9187ee4) I'm guessing, but it looks like something went wrong with that commit.

This hasn't been added to any public release yet. Unrelated.

**Comment by kimjunsung04** [comment]
> 

I haven't checked that yet, thanks

**Comment by kimjunsung04** [comment]
> > [0372ad6#diff-ab27ab8b1311587b5c719143d0bd197a4e311e0dd073ba820f04b608b9187ee4](https://github.com/SkriptLang/Skript/commit/0372ad611477b08a1b285bf8d655e643ad19b086#diff-ab27ab8b1311587b5c719143d0bd197a4e311e0dd073ba820f04b608b9187ee4) I'm guessing, but it looks like something went wrong with that commit.
> 
> This hasn't been added to any public release yet. Unrelated.

I haven't checked that yet, thanks

**Comment by kimjunsung04** [comment]
> ViaVersion

I am not using ViaVersion.

**Comment by kimjunsung04** [comment]
> I cannot replicate. Is everyone using ViaVersion? If so what version is your client?
> 
> On a 1.19.2 client and 1.19.2 server with or without ViaVersion I cannot replicate.
> 
> ViaVersion is designed for lower minecraft versions like 1.9 server to add 1.9+ support. It really serves no purpose being on a server that is the latest (1.19.2) there is even a warning in console from ViaVersion stating this.
> 
> Are you also using ViaBackwards? Because we need to know these things too.
> 
> Video and Server .jar don't work.

I am not using ViaVersion.

**Comment by TheLimeGlass** [comment]
> > I cannot replicate. Is everyone using ViaVersion? If so what version is your client?
> > On a 1.19.2 client and 1.19.2 server with or without ViaVersion I cannot replicate.
> > ViaVersion is designed for lower minecraft versions like 1.9 server to add 1.9+ support. It really serves no purpose being on a server that is the latest (1.19.2) there is even a warning in console from ViaVersion stating this.
> > Are you also using ViaBackwards? Because we need to know these things too.
> > Video and Server .jar don't work.
> 
> I am not using ViaVersion.

Send your plugin list, and how you created the item and what you did to replicate the issue.

**Comment by sovdeeth** [comment]
Can replicate on 1.19.2 paper, 2.6.3 Skript, fresh install.

Used code:
```
on inventory click:
	broadcast "%name of event-item%"
on inventory click:
	broadcast "%event-item%"
```

Created item with effect command `!set name of player's tool to "test"`
Created item with anvil rename
 All broadcasts were none.
![image](https://user-images.githubusercontent.com/10354869/184595236-02283fc2-16ad-451b-81a2-fe200aa3ba7e.png)

![image](https://user-images.githubusercontent.com/10354869/184595262-49f1b399-14d6-49fc-9a2b-e1d5d3b52a56.png)

Used server:
https://drive.google.com/file/d/1Z029Vo8dTrxB8ben_L70O_OaIiQ4BSDo/view?usp=sharing

**Comment by kimjunsung04** [comment]
> > I am not using ViaVersion.
> 
> Send your plugin list, and how you created the item and what you did to replicate the issue.

The plugin only uses Skript and the bucket uses purpur
In the above state, on click inventory at the event
Calling event-item will result in <none>.

**Comment by ZConnorZ** [comment]
I also discovered this, but solved it by replacing event-item with event-slot, including 'index of event-slot'

**Comment by TheLimeGlass** [comment]
Fixed in 2.6.4, but there is more issues we're adding relating to this issue.
